catroid/src/org/catrobat/catroid/ui/dialogs/NewSpriteDialog.java

```diff
@@ -181,6 +181,14 @@ private AlertDialog createDialogStepTwo(AlertDialog.Builder dialogBuilder) {
 		return dialog;
 	}
 
+	private String getFileNameFromPath(String path) {
+		File temp = new File(path);
+		String result = temp.getName();
+		int point = result.indexOf('.');
+		result = result.substring(0, point);
+		return result;
+	}
+
```

Comment:
According to our [build.gradle file](https://github.com/Catrobat/Catroid/blob/develop/build.gradle#L75), we are using guava which already provides a function for that[1]. Also take a quick peek at the source code of the function. It will show you a little :bug: your code may have introduced.
[1] http://docs.guava-libraries.googlecode.com/git/javadoc/com/google/common/io/Files.html#getNameWithoutExtension%28java.lang.String%29